Library/ViewModels/CommentTableViewFooterViewModel.swift

```diff
@@ -17,15 +26,26 @@ public protocol CommentTableViewFooterViewModelType {
 public final class CommentTableViewFooterViewModel: CommentTableViewFooterViewModelType,
   CommentTableViewFooterViewModelInputs, CommentTableViewFooterViewModelOutputs {
   public init() {
-    self.shouldShowActivityIndicator = self.loadingIndicatorSignal
+    let state = self.stateProperty.signal.skipNil()
+
+    self.activityIndicatorHidden = state.map { $0 == .activity }.negate()
+    self.retryButtonHidden = state.map { $0 == .error }.negate()
+    self.rootStackViewHidden = state.map { $0 == .hidden }
+    self.bottomInsetHeight = state.map { state -> Int in
+      if case .error = state { return 4 }
```

Comment:
Why the explicit distinguishing of 2 and 4 in this context? Might be useful to have an inline comment somewhere here.

Reply:
This logic here could be a good cause to have different footer view for both state, as the logic seem to be managing the dynamic layout of both state by adjusting the bottom inset.

Reply:
Oh, it's because with the button we're aligning the content to `top` and this doesn't seem to correctly increase the space below the button so we need to handle this manually here 🤷‍♂️  I can add a comment.